src/timeline/app.js

```diff
@@ -51,32 +64,29 @@ $(document).ready(function () {
 
     track_controls.height(new_track_height);
     $("#scrolling_tracks").height(new_track_height);
-    body_object.scope().playhead_height = $("#track-container").height();
-    $(".playhead-line").height(body_object.scope().playhead_height);
+      body_object.scope().playhead_height = $("#track-container").height();
+      $(".playhead-line").height(body_object.scope().playhead_height);
   });
 
-  // Check for Qt Integration
-  if (typeof timeline !== "undefined") {
-    timeline.qt_log("Qt Found!");
-    body_object.scope().enableQt();
-    timeline.page_ready();
-    body_object.scope().setThumbAddress(timeline.get_thumb_address());
-  }
-
-  // Manually trigger the window resize code (to verify it runs at least once)
-  $(window).trigger("resize");
-
   // Bind to keydown event (to detect SHIFT)
   body_object.keydown(function (event) {
     if (event.which === 16) {
+      if (timeline) {
+        timeline.qt_log("Shift pressed!");
+      }
       body_object.scope().shift_pressed = true;
     }
   });
 
   body_object.keyup(function () {
     if (body_object.scope().shift_pressed) {
+      if (timeline) {
+        timeline.qt_log("Shift released!");
+      }
       body_object.scope().shift_pressed = false;
```

Comment:
I had a think about this. It _really_ needs to check the key value in both directions. Otherwise, say you press <kbd>Shift</kbd>+<kbd>←</kbd>, then release _only_ the <kbd>←</kbd> key. They KeyUp event will trigger, `shift_pressed` will get incorrectly unset, and now the code has _no idea_ you're still holding down <kbd>Shift</kbd>.

If `.keyup()` checks `event.which`, then it can just force `shift_pressed` into the new state without checking it, and the only case that won't be properly accounted for is, say, pressing both <kbd>Shift</kbd> keys at once and then releasing only one of them. But that's pretty crazy and not something we need to worry about.

```suggestion
  body_object.keyup(function () {
    if (event.which === 16) {
      if (timeline) {
        timeline.qt_log("Shift released!");
      }
      body_object.scope().shift_pressed = false;
```

Reply:
@jonoomph HAHAHAHA! Literally **part of Qt's documentation for their own `QKeyEvent`** (what a weird coincidence):

![image](https://user-images.githubusercontent.com/538020/88280810-84a7f200-ccb4-11ea-975e-af34ab0054eb.png)